modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/FindRowVersion.java

```diff
@@ -58,35 +58,35 @@ class FindRowVersion implements PageMemoryTraversal<RowVersionFilter> {
     }
 
     @Override
-    public long consumePagePayload(long link, long pageAddr, DataPagePayload payload, RowVersionFilter arg) {
-        if (!rowVersionFounded) {
-            long nextLink = readPartitionless(partitionId, pageAddr, payload.offset() + RowVersion.NEXT_LINK_OFFSET);
-
-            if (arg.apply(link, pageAddr + payload.offset())) {
-                rowVersionFounded = true;
-
-                rowLink = link;
-                rowTimestamp = HybridTimestamps.readTimestamp(pageAddr, payload.offset() + RowVersion.TIMESTAMP_OFFSET);
-                rowNextLink = nextLink;
-
-                if (!loadValueBytes) {
-                    rowValueSize = PageUtils.getInt(pageAddr, payload.offset() + RowVersion.VALUE_SIZE_OFFSET);
-
-                    return STOP_TRAVERSAL;
-                } else {
-                    return readRowVersionValue.consumePagePayload(link, pageAddr, payload, null);
-                }
-            } else {
-                return nextLink;
-            }
-        } else {
+    public long consumePagePayload(long link, long pageAddr, DataPagePayload payload, RowVersionFilter filter) {
+        if (rowVersionFound) {
+            return readRowVersionValue.consumePagePayload(link, pageAddr, payload, null);
+        }
+
+        long nextLink = readPartitionless(partitionId, pageAddr, payload.offset() + RowVersion.NEXT_LINK_OFFSET);
+
+        if (!filter.apply(link, pageAddr + payload.offset())) {
+            return nextLink;
+        }
+
+        rowVersionFound = true;
+
+        rowLink = link;
+        rowTimestamp = HybridTimestamps.readTimestamp(pageAddr, payload.offset() + RowVersion.TIMESTAMP_OFFSET);
+        rowNextLink = nextLink;
+
+        if (loadValueBytes) {
             return readRowVersionValue.consumePagePayload(link, pageAddr, payload, null);
         }
+
+        rowValueSize = PageUtils.getInt(pageAddr, payload.offset() + RowVersion.VALUE_SIZE_OFFSET);
```

Comment:
btw, you don't read size if you need to load the value. What's the reason for such behavior? Don't we need a valid value for it?

Reply:
Not quite understood, I read this field to understand whether this **RowVersion** is a tombstone or not without copying the array of values ​​of this version.

Reply:
I'm literally asking you, why don't you fill the field "rowValueSize" when "loadValueBytes" is true. Can you please reply?

Reply:
If `loadValueBytes` is `true`, then this field will be read at  `org.apache.ignite.internal.storage.pagememory.mv.ReadRowVersionValue`.